lib/dotenv.rb

```diff
@@ -94,13 +94,20 @@ def restore(env = @diff&.a, safe: Thread.current == Thread.main)
   # Update `ENV` with the given hash of keys and values
   #
   # @param env [Hash] Hash of keys and values to set in `ENV`
-  # @param overwrite [Boolean] Overwrite existing `ENV` values
+  # @param overwrite [Boolean|:warn] Overwrite existing `ENV` values
   def update(env = {}, overwrite: false)
     instrument(:update) do |payload|
       diff = payload[:diff] = Dotenv::Diff.new do
         ENV.update(env.transform_keys(&:to_s)) do |key, old_value, new_value|
           # This block is called when a key exists. Return the new value if overwrite is true.
-          overwrite ? new_value : old_value
+          case overwrite
+          when :warn
+            warn "Warning: Not overwriting ENV['#{key}'] with '#{new_value}' (is '#{old_value}')"
```

Comment:
Can this use the logger instead of `warn`? I am also hesitant about printing the values because they could contain secrets.

Reply:
yeah good point, keys should be enough